packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts

```diff
@@ -457,6 +460,21 @@ export class OdspDocumentStorageService implements IDocumentStorageService {
 
                         retrievedSnapshot = await cachedSnapshotP;
 
+                        // In order to decrease the number of times we have to execute a snapshot refresh,
+                        // if this is the summarizer and we have a cache entry but it is past the defaultFirstSummaryCacheExpiryTimeout,
+                        // force the network retrieval instead as there might be a more recent snapshot available.
+                        if (retrievedSnapshot !== undefined &&
+                            this.hostPolicy.summarizerClient &&
+                            this.lastCacheEntryTime !== undefined) {
+                            const cacheAge = Date.now() - this.lastCacheEntryTime;
```

Comment:
We are calculating cache age twice now. Note that if `snapshotCachedEntry.cacheEntryTime` is `undefined`, that means the snapshot is older than the code itself. 

Here's the other calculation.
`const age = Date.now() - (snapshotCachedEntry.cacheEntryTime ?? (Date.now() - 30 * 24 * 60 * 60 * 1000));`

Reply:
Yes, the idea is to forcibly update the cache if a summarizer kicks in and it has been more than 60 seconds the cache has been updated. 
Notice this will only happen if this.firstVersionCall && count === 1 && (blobid === null || blobid === this.documentId) + IsSummarizer  and we did retrieve a snapthot.

Reply:
We are already recording cacheAge `props["cacheEntryAge"] = age`